sql-plugin/src/main/scala/org/apache/spark/sql/rapids/predicates.scala

```diff
@@ -66,6 +67,125 @@ case class GpuAnd(left: Expression, right: Expression) extends CudfBinaryOperato
 
   override def binaryOp: BinaryOp = BinaryOp.NULL_LOGICAL_AND
   override def astOperator: Option[BinaryOperator] = Some(ast.BinaryOperator.NULL_LOGICAL_AND)
+
+  protected def filterBatch(
```

Comment:
Rather than copying all this code, I expected `GpuAnd` to use the existing `GpuConditionalExpression` trait.

Reply:
Yes absolutely agree.
I was thinking to create a new trait for helpers related to the ColumnVector since those methods are not really specific to "ConditionalExpression". WDYT?

Reply:
Maybe, although it starts to get outside the scope of the PR a bit.  If we are moving stuff around, `filterBatch` should be part of `GpuFilter` which does something almost identical to this method.

Reply:
I could not add `GpuConditionalExpression ` out of the box because of conflicts between the traits tree. So, I pulled the generic methods into a new trait that can be extended by the GpuAnd/GpuOr.
Another approach would be to move that to Helper object, but I did not like that because it would cause more scattered changes in `conditionalExpressions.scala`